src/search/vectorStoreManager.ts

```diff
@@ -52,6 +59,10 @@ export default class VectorStoreManager {
       ) {
         await this.eventHandler.updateExcludedFiles();
       }
+    };
+
+    subscribeToSettingsChange(() => {
+      this.initializationPromise = reinitialize();
```

Comment:
we need to wait for initialization again when settings change causes the singleton to reinitialize